### PR TITLE
Add model support to Menu GateFilter

### LINKED ISSUE #125: QUESTION: Passing a model to a Gate
Is there a way to pass a model to a gate when building the menu items? I have tried a few different things but none seem to work.

**Menu Example**

```php
$event->menu->add([
    'header' => 'HEADER', 
    'can' => 'update' // Doesn't seem like you can pass a model here...
]);
```

**Laravel Documentation**

```php
if (Gate::forUser($user)->allows('update', $model)) {
    // The user can update the model...
}
```

The only way I managed to get it to work was to edit the GateFilter.php and add a model property to the item and pass it if set.

**Modified GateFilter.php**

```php
protected function isVisible($item)
{
    if (isset($item['model'])) {
        return ! isset($item['can']) || $this->gate->allows($item['can'], $item['model']);
    }
    return ! isset($item['can']) || $this->gate->allows($item['can']);
}
```

Am I missing something?

### PR DESCRIPTION
**Pull request by amnoth**
Pass a model to the Gate for authorizing actions.

**Larvel Documentation:**
```php
if (Gate::allows('update', $model)) {
    // The current user can update the post...
}
```

**Usage:**

```php
$event->menu->add([
    'header' => 'HEADER', 
    'can' => 'update',
    'model' => $model
]);
```

closes #125

### CODE REVIEW

File: src/Menu/Filters/GateFilter.php [hunk 1]
```diff
@@ -25,6 +25,10 @@ public function transform($item, Builder $builder)
 
     protected function isVisible($item)
     {
+        if (isset($item['model'])) {
+            return ! isset($item['can']) || $this->gate->allows($item['can'], $item['model']);
+        }
+
```
Comment (jeroennoten) [inline src/Menu/Filters/GateFilter.php]: Please remove the 8 spaces at this empty line.

File: README.md [hunk 2]
```diff
@@ -209,7 +209,8 @@ Use the `can` option if you want conditionally show the menu item. This integrat
     [
         'text' => 'Add new post',
         'url' => 'admin/blog/new',
-        'can' => 'add-blog-post'
+        'can' => 'add-blog-post',
+        'model' => $model
```
Comment (jeroennoten) [inline README.md]: Adding this here may suggest that you can use models in the `adminlte.php` configuration file, which is not possible. I think this `'model'` documentation should be explained separately.

File: src/Menu/Filters/GateFilter.php [hunk 3]
```diff
@@ -25,6 +25,10 @@ public function transform($item, Builder $builder)
 
     protected function isVisible($item)
     {
+        if (isset($item['model'])) {
```
Comment (jeroennoten) [inline src/Menu/Filters/GateFilter.php]: It is better to start this method with:

```
if (!isset($item['can'])) {
    return true
}
```

This way, we do not need the `! isset($item['can']) || ` part in the return statements.

File: src/Menu/Filters/GateFilter.php (hunk 1 above)
Comment (amnoth) [reply src/Menu/Filters/GateFilter.php]: Sorry, I'm a bit new to this. I originally only saw this and pushed a change.

File: src/Menu/Filters/GateFilter.php (hunk 3 above)
Comment (amnoth) [reply src/Menu/Filters/GateFilter.php]: Sounds good to me. :D

**Comment by jeroennoten** [comment]
Thanks you very much! Congratulations to your first merged PR ;-)

**Pull request merged**